content/accessibility/truncation.mdx

```diff
@@ -0,0 +1,95 @@
+
+### Explicit truncation
+
+Explicit truncation means that a certain text is truncated and it includes an actual control in the interface that can be clicked on or pressed with the keyboard, letting the user expand what is displayed.
+
+## General Rules
+
+### Focusable elements
+
+As a general rule we should not use implicit truncation text that can contain focusable elements.
+
+Some of the potential issues are:
+- the focusable elements could be completely hidden and they will randomly appear when the user navigates into them.
+- the focus seems to disappear.
+- focusable elements can be truncated in the middle.
+- "unsecure" links can be focused via keyboard whith no visual focus.
+
+### Security
+
+We don't want to ever truncate something that causes any potential security concerns.
+For example, truncating a URL is a security bad practice, because the user doesn't have insight into where the link might be going.
+
+### Untrue information
+
+In some situations we don't control in which part of the text the truncation is happening, so we wan't to avoid to truncate a text if that could be considered as 
+"untrue information.
+Ideally never truncate in the middle of a word, and even more ideally never truncate in the middle of a sentence or paragraph.
+
+For example, if a text include mentions, truncation can happend at the middle of one, confusing the user who's reading the text.
+Or we truncate the name of a company, and we don't control where the truncation is happening, it might be that several companies with the similar beginning at their name have the same visual truncation, and that would be very confusing for the user.
+
+## Truncation options
+
+### CSS Truncation
+
+CSS provides mechanisms allowing you to visually hide or truncate content without needing to touch the underlying HTML.
+CSS truncation will not removed the truncated text from the DOM, so it could look unreadable for a sighted user, but screen reader users will still be able to read it.
+
+This method is purely controlled by the browser, and gives us no power on deciding where to truncate the text.
+
+We should not truncate large bodies of text using css. but there are times when we can truncate using CSS. If you're working with a user interface control where the truncated text is not controlled by designers and developers, for example a single line of text where cannot fit all the text on it, actually CSS truncation is the right thing to do because you're gonna show the full thing in an accessible tooltip.
+
+If you're working with a user interface control where GitHub designers and developers have decided exactly what text it will have and exactly what its purpose is, meaning that is a generated conten and it's known ahead of time, then we should avoid truncating it.
```

Comment:
```suggestion
If you're working with a user interface control where GitHub designers and developers have decided exactly what text it will have and exactly what its purpose is, meaning that is a generated content and it's known ahead of time, then we should avoid truncating it.
```